index.js

```diff
@@ -1,48 +1,53 @@
 'use strict'
 const json = require('fast-json-stable-stringify')
-const CID = require('cids')
 const isCircular = require('is-circular')
 const transform = require('lodash.transform')
-const codecInterface = require('@ipld/codec-interface')
 
-const _encode = obj => transform(obj, (result, value, key) => {
-  if (CID.isCID(value)) {
-    result[key] = { '/': value.toBaseEncodedString() }
-  } else if (Buffer.isBuffer(value)) {
-    result[key] = { '/': { base64: value.toString('base64') } }
-  } else if (typeof value === 'object' && value !== null) {
-    result[key] = _encode(value)
-  } else {
-    result[key] = value
-  }
-})
+module.exports = multiformats => {
+  const { CID, bytes, multibase } = multiformats
+  if (!multibase.has('base64')) throw new Error('Multibase must include base64')
```

Comment:
I decided not to do this at init time because it imposes strict ordering on the user. It's going to be less error prone but I can imagine this pattern leading to really awkward circular dependency problems if we encourage it. Not a -1 but I'd be just as comfortable leaving it out and letting an `encode()` fail with a meaningful error ('unknown base, did you forget to add it?').

Reply:
this is a good point, i might back it out